sdk/nodejs/introspector/registry/registry.ts

```diff
@@ -170,7 +170,7 @@ export class Registry {
     return (
       fnStr
         .slice(fnStr.indexOf("(") + 1, fnStr.indexOf(")"))
-        .match(/([^\s,]+)/g) ?? []
+        .match(/\b(\w+)\b(?:(?=\s*[:=?]|,\s*|$))/g) ?? []
```

Comment:
Can't you have the right order from the introspection phase?

Reply:
No, it's not possible because prototype are undefinied and the scanner isn't called in case of an invokation

Reply:
So your problem is not knowing how to have access to the introspection data when `parentName` is not empty?

Reply:
Exactly

Reply:
I'll help you in a follow up then. This should be possible to do.